forms/gridfield/GridField.php

```diff
@@ -177,6 +174,11 @@ public function getConfig() {
 	public function setConfig(GridFieldConfig $config) {
 		$this->config = $config;
 
+		if (!$this->config->getComponentByType('GridState_Component')) {
+			$this->config->addComponent(new GridState_Component());
+			$this->state = new GridState($this);
```

Comment:
Looks like online edit works :) 
@tractorcow and @dhensby I moved the line back to constructor.